Sources/MultipartKit/MultipartParserError.swift

```diff
@@ -1,7 +1,24 @@
 /// Technical parsing error, such as malformed data or invalid characters.
 /// This is mainly used by ``MultipartParser``.
-package enum MultipartParserError: Swift.Error, Equatable {
-    case invalidBoundary
-    case invalidHeader(reason: String)
-    case invalidBody(reason: String)
+public struct MultipartParserError: Swift.Error, Equatable {
+    public let reason: Reason
+
+    private init(reason: Reason) {
+        self.reason = reason
+    }
+
+    public enum Reason: Equatable {
+        case invalidBoundary
+        case invalidHeader(reason: String)
+        case invalidBody(reason: String)
+    }
+
+    public static let invalidBoundary = MultipartParserError(reason: .invalidBoundary)
+    
+    public static func invalidHeader(_ reason: String) -> MultipartParserError {
+        return MultipartParserError(reason: .invalidHeader(reason: reason))
```

Comment:
since we changed the initializer:
```suggestion
        return MultipartParserError(.invalidHeader(reason: reason))
```